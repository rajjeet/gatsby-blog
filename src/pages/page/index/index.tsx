import React from 'react';
import { graphql } from 'gatsby';
import Layout from '../../../components/layout';
import PostListing from '../../../components/post-listing';
import Seo from '../../../components/seo';
import ProjectListing from '../../../components/project-listing';
import { TPostEdge, TImage } from '../../../templates/category-page/types';

export type TProjectEdge = {
  node: {
    frontmatter: {
      title: string;
      description: string;
      thumbnail: TImage;
      techStackTags: {label: string; type: string}[];
      links: {label: string; value: string}[];
    };
    fields: {
      slug: string;
    };
    body: string;
  };
};

export type TProps = {
  data: {
    posts: {
      edges: TPostEdge[];
    };
    projects: {
      edges: TProjectEdge[];
    };
  };
};

const IndexPage: React.FC<TProps> = ({ data }) => (
  <Layout>
    <Seo />
    <ProjectListing
      projects={data.projects.edges}
      heading="Side Projects"
    />
    <PostListing
      posts={data.posts.edges}
      heading="Latest Posts"
    />
  </Layout>
);

export default IndexPage;

export const query = graphql`
    {
        posts: allMdx(sort: {fields: [frontmatter___date], order: DESC}, filter: {frontmatter: {draft: {ne: true}}, fields: {contentType: {eq: "post"}}}, limit: 3) {
            edges {
                node {
                    ...PostListingMarkdownFragment
                }
            }
        }
        projects: allMdx(limit: 2, sort: {fields: frontmatter___date, order: DESC}, filter: {fields: {contentType: {eq: "project"}}}) {
            edges {
                node {
                    ...ProjectListingMarkdownFragment
                }
            }
        }
    }
`;
