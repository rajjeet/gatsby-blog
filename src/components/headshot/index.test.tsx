import * as React from 'react';
import { render } from '@testing-library/react';
import { Headshot } from './index';

describe('Headshot', () => {
  it('should render', () => {
    const { asFragment } = render(<Headshot />);
    expect(asFragment()).toMatchSnapshot();
  });
  it('should show the headshot of site author', () => {
    const { getByAltText } = render(<Headshot />);
    expect(getByAltText('Headshot of site author')).toBeDefined();
  });
});
