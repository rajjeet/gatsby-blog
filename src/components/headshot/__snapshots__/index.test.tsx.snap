// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Headshot should render 1`] = `
<DocumentFragment>
  .c0 {
  margin: 1rem;
  width: 170px;
  border-radius: 50%;
  border: .33rem solid white;
}

<div
    class="c0 gatsby-image-wrapper"
    style="position: relative; overflow: hidden;"
  >
    <div
      style="width: 100%; padding-bottom: 100%;"
    />
    <picture>
      <source
        sizes=""
        srcset=""
      />
      <img
        alt="Headshot of site author"
        loading="lazy"
        sizes=""
        src="https://via.placeholder.com/250x180"
        srcset=""
        style="position: absolute; top: 0px; left: 0px; width: 100%; height: 100%; object-fit: cover; object-position: center; opacity: 0; transition: opacity 500ms;"
      />
    </picture>
    <noscript>
      &lt;picture&gt;&lt;source srcset="" /&gt;&lt;img loading="lazy" src="https://via.placeholder.com/250x180" alt="Headshot of site author" style="position:absolute;top:0;left:0;opacity:1;width:100%;height:100%;object-fit:cover;object-position:center"/&gt;&lt;/picture&gt;
    </noscript>
  </div>
</DocumentFragment>
`;
